sycl/test-e2e/format.py

```diff
@@ -389,12 +389,22 @@ def map_result(features, code):
                 elif code is lit.Test.FAIL:
                     code = lit.Test.XFAIL
             return code
+        def check_leak(output):
+            keyword_found = False
+            for line in output.splitlines():
+                if keyword_found and "LEAK" in line:
```

Comment:
Just FYI, we already have a lit var  `%{l0_leak_check}`  which a bunch of the tests still use (instead of `UR_L0_LEAKS_DEBUG` ).  It gets replaced by the env var in the final invocation, but  I'm not sure if this python script here will detect.

I think that `l0_leak_check` directive is no longer needed and probably could be removed.  It's in hundreds of tests right now.

Reply:
I think the tests that set the env var (either directly or by using %{l0_leak_check}) already check for absence of "LEAK" (--implicit-check-not=LEAK) so we don't have to check it again here. As far as I understand this is only needed when the user decides to run all the tests with leak checking (e.g. passing --param ur_l0_leaks_debug=1 to lit).